code/modules/events/aurora_caelus.dm

```diff
@@ -12,51 +12,68 @@
-	priority_announce("[station_name()]: A harmless cloud of ions is approaching your station, and will exhaust their energy battering the hull. Nanotrasen has approved a short break for all employees to relax and observe this very rare event. During this time, starlight will be bright but gentle, shifting between quiet green and blue colors. Any staff who would like to view these lights for themselves may proceed to the area nearest to them with viewing ports to open space. We hope you enjoy the lights.",
+	priority_announce("[station_name()]: A harmless cloud of ions is approaching your station, and will exhaust their energy battering the hull. Nanotrasen has approved a short break for all employees to relax and observe this very rare event. During this time, starlight will be bright but gentle, shifting between quiet green, blue and purple colors. Any staff who would like to view these lights for themselves may proceed to the area nearest to them with viewing ports to open space. We hope you enjoy the lights.",
 	sound = 'sound/misc/notice2.ogg',
 	sender_override = "Nanotrasen Meteorology Division")
+	starttime = world.timeofday
 	for(var/V in GLOB.player_list)
 		var/mob/M = V
-		if((M.client.prefs.toggles & PREFTOGGLE_SOUND_MIDI) && is_station_level(M.z))
-			M.playsound_local(M, 'sound/ambience/aurora_caelus.ogg', 20, FALSE, pressure_affected = FALSE)
+		if(is_station_level(M.z))
+			SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "star_gazing", /datum/mood_event/witnessed_starlight)
+			if(M.client.prefs.toggles & PREFTOGGLE_SOUND_MIDI)
+				M.playsound_local(M, 'sound/ambience/aurora_caelus.ogg', 20, FALSE, pressure_affected = FALSE)
+	for(var/atom/window in GLOB.station_windows)
+		for(var/turf/T as() in RANGE_TURFS(1, window))
+			if(isspaceturf(T) && !(T.starlit))
+				T.set_light(1.5, 0.1, aurora_colors[7])
+				T.starlit = TRUE
+				affected_turfs += T
 
 /datum/round_event/aurora_caelus/start()
-	for(var/area in GLOB.sortedAreas)
-		var/area/A = area
-		if(initial(A.dynamic_lighting) == DYNAMIC_LIGHTING_IFSTARLIGHT)
-			for(var/turf/open/space/S in A)
-				S.set_light(S.light_range * 3, S.light_power * 0.5)
+	if(!length(affected_turfs))
+		return
+	for(var/atom/A in affected_turfs)
+		A.set_light(2, 0.2, l_color = aurora_colors[8])
 
 /datum/round_event/aurora_caelus/tick()
 	if(activeFor % 5 == 0)
-		aurora_progress++
+		if(aurora_progress < 8)
+			aurora_progress++
+		if(!length(affected_turfs))
+			return
 		var/aurora_color = aurora_colors[aurora_progress]
-		for(var/area in GLOB.sortedAreas)
-			var/area/A = area
-			if(initial(A.dynamic_lighting) == DYNAMIC_LIGHTING_IFSTARLIGHT)
-				for(var/turf/open/space/S in A)
-					S.set_light(l_color = aurora_color)
+		var/light_modifier = 0
+		if(aurora_progress < 5)
+			light_modifier = aurora_progress / 10
+		else
+			light_modifier = 1 - aurora_progress / 10
+		for(var/atom/A in affected_turfs)
+			A.set_light(2 + light_modifier * 4, 0.3, l_color = aurora_color)
+
```

Comment:
the whole codes in tick() here should be in a separate proc, and that proc should be called by ASYNC()